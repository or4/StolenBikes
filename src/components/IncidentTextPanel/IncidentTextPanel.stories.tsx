import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';

import { IncidentTextPanel } from './IncidentTextPanel';

const stories = storiesOf('IncidentTextPanel', module);
stories.addDecorator(withKnobs);

stories.add('IncidentTextPanel', () => {
    return (
        <IncidentTextPanel
            title={getTitleKnobs()}
            description={getDescriptionKnobs()}
            occurredAt={1532152800}
            address={getAddressKnobs()}
        />
    );
});

function getTitleKnobs() {
    return text('Header', 'Stolen 2018 VAUN VELO(black)');
}

function getDescriptionKnobs() {
    return text('Description', 'Stolen from my courtyard');
}

function getAddressKnobs() {
    return text('Address', 'Berlin, 12047, DE');
}
