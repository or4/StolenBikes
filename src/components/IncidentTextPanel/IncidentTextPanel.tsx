import React from 'react';
import styled from 'styled-components';

import { convertFromUnixTimestamp } from 'core/utils/unixTimestamp';
import { media } from 'ui/media';

interface Props {
    title?: string;
    description?: string | null;
    occurredAt?: number;
    address?: string;
}

const Container = styled.div`
    display: flex;
    flex-direction: column;

    ${media.tabletNDesktop`
    	padding: 0 15px;
	`}
`;

const Title = styled.div`
    font-weight: 600;

    ${media.mobile`
		padding-bottom: 5px;
  	`}

    ${media.tabletNDesktop`
		padding-bottom: 10px;
	`}
`;

const Description = styled.div`
    ${media.mobile`
		padding-bottom: 5px;
  	`}

    ${media.tabletNDesktop`
		padding-bottom: 10px;
	`}
`;

const Details = styled.div``;

export const IncidentTextPanel = (props: Props) => {
    const details = getDetails(props.occurredAt, props.address);

    return (
        <Container>
            {props.title && <Title>{props.title}</Title>}
            {props.description && <Description>{props.description}</Description>}
            {details && <Details>{details}</Details>}
        </Container>
    );
};

function getDetails(occurredAt?: number, address?: string): string {
    if (occurredAt && address) {
        return `${dateToString(occurredAt)} - ${address}`;
    }

    if (occurredAt) {
        return dateToString(occurredAt);
    }

    if (address) {
        return address;
    }

    return '';
}

function dateToString(date: number): string {
    return new Date(convertFromUnixTimestamp(date)).toDateString();
}
