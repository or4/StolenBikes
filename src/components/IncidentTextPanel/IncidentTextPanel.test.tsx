import React from 'react';
import { mount } from 'enzyme';

import { IncidentTextPanel } from './IncidentTextPanel';

describe('IncidentTextPanel', () => {
    it('should render correctly', () => {
        const component = <IncidentTextPanel />;

        const container = mount(component);
        expect(container).toMatchSnapshot();
        container.unmount();
    });
});
