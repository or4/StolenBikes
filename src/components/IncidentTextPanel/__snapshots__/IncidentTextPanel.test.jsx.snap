// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncidentTextPanel should render correctly 1`] = `
<IncidentTextPanel>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "gdbsuu",
            "rules": Array [
              "
    display: flex;
    flex-direction: column;
    padding: 0 15px;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa gdbsuu"
      />
    </StyledComponent>
  </styled.div>
</IncidentTextPanel>
`;
